payments-core/src/main/java/com/stripe/android/utils/CustomViews.kt

```diff
@@ -0,0 +1,61 @@
+package com.stripe.android.utils
+
+import android.view.View
+import androidx.core.view.doOnAttach
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.stripe.android.BuildConfig
+import com.stripe.android.view.CardWidgetViewModel
+import com.stripe.android.view.RealCbcEnabledProvider
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+
+internal fun <T : ViewModel> View.doWithViewModel(action: LifecycleOwner.(T) -> Unit) {
+    doOnAttach {
+        val lifecycleOwner = findViewTreeLifecycleOwner()
+        val viewModelStoreOwner = findViewTreeViewModelStoreOwner()
```

Comment:
These two will return `null` if called before the view is attached. The `doWithViewModel` is supposed to prevent this misuse.